### PR TITLE
Explain fields for requiredlabel/annotation CTs

### PR DESCRIPTION
**Pull request by willbeason**

### CODE REVIEW

File: library/general/requiredannotations/template.yaml [hunk 1]
```diff
@@ -18,13 +19,20 @@ spec:
...
                 type: object
                 properties:
                   key:
                     type: string
+                    description: >-
+                      The required annotation.
                   allowedRegex:
                     type: string
+                    description: >-
+                      A regular expression the annotation's value must match. The value
```
Comment (maxsmythe) [inline library/general/requiredannotations/template.yaml:34]: We should probably mention that this field is optional, and that the regex wont be checked if `allowedRegex` is undefined.

Comment (willbeason) [reply library/general/requiredannotations/template.yaml:34]: Done!

File: library/general/requiredannotations/template.yaml [hunk 2]
```diff
@@ -18,13 +19,21 @@ spec:
...
+                      must contain at least one match for the regular expression.
+                      If unset, or set to empty string, matches all values.
```
Comment (maxsmythe) [inline library/general/requiredannotations/template.yaml:36]: nit: I think we can ignore the empty-string case... is an empty string regex one that matches everything?

**Review by maxsmythe — APPROVED** [review]
LGTM

File: library/general/requiredannotations/template.yaml (hunk 2 above)
Comment (willbeason) [reply library/general/requiredannotations/template.yaml:36]: Sort of - the wording is weird. The empty string regex has matches in all existent strings. It doesn't "match" the text (like ".*" would), but it "has matches" in the text.

**Review by julianKatz — APPROVED** [review]
If Max's comment is resolved, LGTM

File: library/general/requiredannotations/template.yaml (hunk 2 above)
Comment (maxsmythe) [reply library/general/requiredannotations/template.yaml:36]: That sounds like it's covered by "must contain at least one match for the regular expression"

For the undefined case, you can prefix the first sentence with "When defined, a regular expression..."

Basically, it'd be good to make it clear that users can ignore this field unless they want to leverage it.

Comment (willbeason) [reply library/general/requiredannotations/template.yaml]: Done.

**Comment by julianKatz** [comment]
ready to merge?

**Review by sozercan — APPROVED** [review]

**Pull request merged**